### ISSUE TITLE
YOLO segmentation on custom dataset 

### ISSUE THREAD

**Issue by kkk2705** [ultralytics/ultralytics#18678]
### Search before asking

- [ ] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi,
I used YOLOv8n and YOLO11n to train my custom dataset of fruits (500 images with a white background under a single light condition). I tried training using both the pretrained .pt file and from scratch using a .yaml file. However, the results show significant noise, and the model is also segmenting surrounding objects incorrectly. Below are some pictures for reference.
![Capture_0](https://github.com/user-attachments/assets/82e79cec-1fda-4f7d-850f-1ba03ecea074)
![Capture_1](https://github.com/user-attachments/assets/8b5d6acf-bac6-42d1-abda-80b91b41c091)
![original image](https://github.com/user-attachments/assets/9f1180ba-40c4-42a0-a4f8-265ecc62441e)

### Additional

How can I reduce noise and incorrect segmentation?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @kkk2705, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it, such as the exact commands or scripts you used for training and validation, along with your dataset configuration file (.yaml). If you're unsure, check out the [custom training documentation](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/) for guidance. 

If this is a custom training ❓ Question, please share additional details to help us assist you better:
1. Your training command or script.
2. The dataset YAML file you've used.
3. Insights into your training setup (learning rate, epochs, batch size, augmentation settings, etc.).
4. The size and characteristics of your dataset, along with any preprocessing methods applied.

### Training Tips
- Verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to ensure optimal outcomes.
- The layout, lighting conditions, and background of your dataset images may strongly influence noise and segmentation accuracy. Explore strategies like increasing your dataset size or introducing diverse lighting and background variations if possible.

### Visual Debugging
To better understand the issue, include clear examples of the segmentation ground truth in comparison to the model's predictions. This can help identify areas needing improvement in preprocessing, augmentation, or parameter tuning.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response. An Ultralytics engineer will assist you further as soon as possible. Thank you for your patience 🙂

**Comment by kkk2705** [comment]
**training script**
import os
from ultralytics import YOLO

# Load the model configuration for your custom dataset
model = YOLO('yolov8n-seg.yaml') 
model.train(data='/content/gdrive/MyDrive/orange/data.yaml', epochs=50, batch=16, verbose=True)

**yaml file**
train: /content/gdrive/My Drive/orange/train/images
val: /content/gdrive/My Drive/orange/valid/images
nc: 2
names: ['Immature', 'Mature']

 **characteristics of dataset**
500 images white background only 
AUGMENATION: Crop: 0% Minimum Zoom, 30% Maximum Zoom
Exposure: Between -5% and +5%.
LIKE BELOW
![66_jpg rf ffbfe1d12581a7f4018818c0132c7ba6](https://github.com/user-attachments/assets/f82367a3-84b7-426a-8f8f-2e6cc9c45bb5)

**Comment by Y-T-G** [comment]
Your model underfitted probably because your dataset is not diverse enough (same or similar image repeated multiple times). Add more diverse data and also background images
